options/options.go

```diff
@@ -426,6 +426,8 @@ func (opts *ToolOptions) AddOptions(extraOpts ExtraOptions) {
 // Parse the command line args.  Returns any extra args not accounted for by
 // parsing, as well as an error if the parsing returns an error.
 func (opts *ToolOptions) ParseArgs(args []string) ([]string, error) {
+	LogSensitiveOptionWarnings(args)
```

Comment:
what's the rationale for iterating over the raw args instead of just usin the parsed args?

Reply:
This is where the order becomes relevant for how we get password values, specifically with adding `ParseConfigFile()` in [the other PR](https://github.com/mongodb/mongo-tools-common/pull/62). From past discussions, we have to call `ParseConfigFile()` before we parse the command line args for us to get the correct command line overriding behavior.

So if we used the parsed args here (i.e. the `*ToolOptions`) it wouldn't be clear whether a password value came from the command line or from `--config`, provided that we want to print a warning only for the former. So we need to look at the raw args directly.

Reply:
why can't `LogSensitiveOptionWarnings` just parse the provided args into the structs and check the struct fields?

Reply:
Oh true, I suppose it doesn't matter how many times we parse into the `*ToolOptions` as long as the order is correct.

The only thing that might be tricky about that is the positional connection string argument. That's handled after all parsing using `opts.setURIFromPositionalArg()`, so parsing into the opts struct won't help with that. We'd still have to loop over the raw args and check if there's a positional connection string.

If that sounds right to you, do you think we should parse into the opts struct and then loop over the raw args just for the connection string? Or should loop and check for everything simultaneously as-is?